prost-build/src/code_generator.rs

```diff
@@ -1160,35 +1248,43 @@ fn build_enum_value_mappings<'a>(
     do_strip_enum_prefix: bool,
     enum_values: &'a [EnumValueDescriptorProto],
 ) -> Vec<EnumVariantMapping<'a>> {
-    let mut numbers = HashSet::new();
     let mut generated_names = HashMap::new();
-    let mut mappings = Vec::new();
+    // Use an insertion-order preserving map here because the enum ordering must be preserved.
+    let mut mappings = IndexMap::<i32, EnumVariantMapping>::new();
```

Comment:
This could be simplified. This is really just a sequence of `iter().unique_by().map().collect()`.